### PR TITLE
LPD-35920 Add attachment to orders - service

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/LPD-35920

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderServiceImpl.java [hunk 1]
```diff
@@ -49,6 +52,20 @@
...
 public class CommerceOrderServiceImpl extends CommerceOrderServiceBaseImpl {
 
+	@Override
+	public FileEntry addAttachment(
+			String externalReferenceCode, long userId, long commerceOrderId,
+			String fileName, InputStream inputStream)
+		throws PortalException {
+
+		_commerceOrderModelResourcePermission.check(
+			getPermissionChecker(), commerceOrderId, ActionKeys.VIEW);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderServiceImpl.java:62]: here I think the check should be done against the UPDATE permission

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/model/impl/CommerceOrderImpl.java [hunk 2]
```diff
@@ -120,6 +148,29 @@ public int getCustomerCommerceOrderIdsCount() {
...
+		}
+
+		return localRepository.fetchFolderByExternalReferenceCode(
+			"order-" + getCommerceOrderId());
+	}
+
+	@Override
+	public LocalRepository getLocalRepository() throws PortalException {
+		Repository repository = RepositoryLocalServiceUtil.fetchRepository(
+			getGroupId(), CPConstants.SERVICE_NAME_PRODUCT);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/model/impl/CommerceOrderImpl.java:164]: I think we should add a name for the ORDER and not use the Product one, please check with other example to determine the pattern

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderServiceImpl.java [hunk 3]
```diff
@@ -154,6 +171,17 @@ public CommerceOrder applyCouponCode(
 			commerceOrderId, couponCode, commerceContext);
 	}
 
+	@Override
+	public void deleteAttachment(long attachmentId, long commerceOrderId)
+		throws PortalException {
+
+		_commerceOrderModelResourcePermission.check(
+			getPermissionChecker(), commerceOrderId, ActionKeys.VIEW);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderServiceImpl.java:179]: UPDATE(?)

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
just these.

Thank you!

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/model/impl/CommerceOrderImpl.java (hunk 2 above)
Comment (smottal) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/model/impl/CommerceOrderImpl.java]: Used CommerceConstants.SERVICE_NAME_COMMERCE_ORDER

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:8202308

**Comment by smottal** [comment]
ci:forward

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:forward

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**